kyo-scheduler/jvm/src/main/scala/kyo/scheduler/Worker.scala

```diff
@@ -0,0 +1,149 @@
+package kyo.scheduler
+
+import Worker.*
+import java.lang.invoke.MethodHandles
+import java.lang.invoke.VarHandle
+import java.util.concurrent.Executor
+import java.util.concurrent.atomic.LongAdder
+import kyo.stats.internal.MetricReceiver
+import scala.util.control.NonFatal
+
+final private class Worker(
+    id: Int,
+    exec: Executor,
+    scheduleTask: (Task, Worker) => Unit,
+    stealTask: Worker => Task,
+    getCurrentCycle: () => Long,
+    clock: InternalClock
+) extends Runnable:
+
+    val a1, a2, a3, a4, a5, a6, a7 = 0L // padding
+
+    @volatile private var running       = false
```

Comment:
This class hasn't changed much but I've made `running` a `volatile` that is updated using `VarHandle` as a micro optimization to avoid the pointer chasing with `AtomicBoolean` since the flag used in the hot path of task execution.